ui/job-view/pushes/PushJobs.jsx

```diff
@@ -96,6 +108,19 @@ class PushJobs extends React.Component {
     jobInstance.toggleRunnableSelected();
   };
 
+  handleGroupSelectAllRunnableClick = (ev, groupNode) => {
+    const { toggleSelectedRunnableJob } = this.props;
+
+    for (const jobButton of groupNode.querySelectorAll('.runnable-job-btn')) {
+      const jobInstance = findInstance(jobButton);
+      if (!jobInstance) {
+        continue;
+      }
+      toggleSelectedRunnableJob(jobInstance.props.job.signature);
+      jobInstance.toggleRunnableSelected();
```

Comment:
What the `[All]` button does is to toggle the selected state for all jobs.

so, if all jobs are selected, or all jobs are not selected, it works as select/deselect all.

if only some jobs are selected, they are de-selected, and all other jobs are selected.

If this isn't okay, other options are:
  * if less than 50% are selected, select all. otherwise de-select all
  * if the first job is not selected, select all. otherwise de-select all

Reply:
Can you implement a solution that doesn't manipulate the DOM directly? Its generally frowned upon to do so with React, although I realize sometimes its unavoidable (and yes, there is some old code that does it here that needs to be updated). Wondering if a ref on the buttons would work.